packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -731,6 +735,15 @@ export class TerminalFrontendContribution implements FrontendApplicationContribu
         menus.registerMenuAction([...TerminalMenus.TERMINAL_CONTEXT_MENU, '_4'], {
             commandId: TerminalCommands.KILL_TERMINAL.id
         });
+
+        menus.registerSubmenu(TerminalMenus.TERMINAL_CONTRIBUTIONS, '', {
+            role: CompoundMenuNodeRole.Group
+        });
+
+        menus.registerSubmenu(TerminalMenus.TERMINAL_TITLE_CONTRIBUTIONS, '', {
+            role: CompoundMenuNodeRole.Group,
+            when: 'tabContextWidgetId =~ /^terminal-[0-9]+$/'
```

Comment:
This regex might be too restrictive. I have the contributed menu on the title when this is a SHELL terminal, but I do not have it for the Javascript Debug Terminal. In VS Code, I have the menu contribution for both of them.

Reply:
should be fixed now. Though if you have a better idea i'd be happy. Setting the `isTerminalTab` by looking if there is a field in a widget is probably not the nicest solution. I just can't think of anything else